### PR TITLE
fix: add onboarding roster guard and skill overwrite warning

### PR DESCRIPTION
**Pull request by diberry**
### Item 1: Onboarding flow production guard
After `createTeam()` in `finalizeCast()`, verify the roster is populated before re-dispatching to the coordinator. If empty:
- Cleans up `.init-prompt` (prevents auto-retry loops)
- Shows user-visible warning
- Clears activity hint and processing state

**File:** `packages/squad-cli/src/cli/shell/index.ts`

### Item 2: Skill upgrade overwrite warning
During `squad upgrade`, warn when a customized built-in skill is about to be overwritten. Fires in both code paths:
- **Manifest upgrade loop** (full version upgrade)
- **`syncAllSkills()`** (already-current ensure checks)

Normalizes line endings before comparison to avoid false positives on Windows.

**File:** `packages/squad-cli/src/cli/core/upgrade.ts`

### Item 3: Regression tests for finalizeCast roster guard
Added 7 regression tests in `test/init-autocast.test.ts` under a new `finalizeCast roster guard — empty roster prevents dispatch` describe block, covering:
- Empty roster after `createTeam()` → `.init-prompt` deleted (prevents auto-retry loop)
- Empty roster after `createTeam()` → dispatch aborted (early return)
- Populated roster → `.init-prompt` preserved, dispatch proceeds normally
- Edge cases: header-only table, absent `.init-prompt`, completely empty `team.md`

**File:** `test/init-autocast.test.ts`

### Notes
- Pre-existing build error in `start.ts` is unrelated
- Both fixes are additive guards/warnings — no behavior changes
- Regression tests ensure the empty-roster dispatch guard cannot be silently removed in future refactors
- Changeset included

**Requested by:** Dina Berry

### CODE REVIEW

File: packages/squad-cli/src/cli/shell/index.ts [hunk 1]
```diff
@@ -986,6 +986,26 @@ export async function runShell(): Promise<void> {
...
+      }
+      shellApi?.addMessage({
+        role: 'system',
+        content: '⚠ Team creation completed but roster is empty — skipping dispatch. Check .squad/team.md.',
+        timestamp: new Date(),
+      });
+      shellApi?.setActivityHint(undefined);
+      shellApi?.setProcessing(false);
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (diberry) [reply packages/squad-cli/src/cli/shell/index.ts:1007]: @copilot apply changes based on [this feedback](https://github.com/bradygaster/squad/pull/867#discussion_r3037618358)

File: packages/squad-cli/src/cli/core/upgrade.ts [hunk 2]
```diff
@@ -431,6 +431,20 @@ export function ensureCastingDefaults(dest: string, templatesDir?: string): stri
...
+ */
+function warnIfSkillCustomized(srcPath: string, destPath: string, sourceName: string): void {
+  if (!storage.existsSync(destPath)) return;
+  const existing = (storage.readSync(destPath) ?? '').replace(/\r\n/g, '\n');
+  const template = (storage.readSync(srcPath) ?? '').replace(/\r\n/g, '\n');
+  if (existing && existing !== template) {
+    const skillName = sourceName.split('/')[1] ?? sourceName;
+    warn(`Skill '${skillName}' has been customized — overwriting with built-in version`);
+  }
+}
```
[the remark this answers is not in the document]
Comment (diberry) [reply packages/squad-cli/src/cli/core/upgrade.ts:446]: @copilot apply changes based on [this feedback](https://github.com/bradygaster/squad/pull/867#discussion_r3037618367)

[the remark this answers is not in the document]
Comment (diberry) [reply packages/squad-cli/src/cli/core/upgrade.ts:446]: Tests already present in commit ea53d68f. Four cases covered in `test/cli/upgrade.test.ts`: (1) modified skill content triggers warning, (2) CRLF-only differences do NOT trigger warning, (3) identical content does NOT trigger warning, (4) full version upgrade path with customized skill triggers warning. All 28 tests pass.

**Review by tamirdresher — APPROVED** [review]
Approved by @tamirdresher — reviewed diff, CI green, changes are clean and well-tested. cc @diberry

**Comment by bradygaster** [comment]
Closed by #theSquadsquad triage — verified fixed in v0.9.4.
